.github/workflows/release.yml

```diff
@@ -0,0 +1,29 @@
+name: Release to npm
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 'lts/*'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build package (if necessary)
+        run: npm run build
+
+      - name: Publish to npm
+        run: npm publish --tag ${{ github.event.release.tag_name }}
```

Comment:
I think you just want to edit this file: `.github/workflows/npm-publish.yml`

It is already hooked up